test/e2e/autoscaling/dns_autoscaling.go

```diff
@@ -99,6 +99,8 @@ var _ = SIGDescribe("DNS horizontal autoscaling", func() {
 	// This test is separated because it is slow and need to run serially.
 	// Will take around 5 minutes to run on a 4 nodes cluster.
 	It("[Serial] [Slow] kube-dns-autoscaler should scale kube-dns pods when cluster size changed", func() {
```

Comment:
The problem is that for multizonal and regional clusters, we are passing only one of the underlying MIGs in NodeInstanceGroups var (or to be more specific only migs from a single zone are passed there). Also NumNodes is set to the number of nodes in a single zone (similarly to what we'are passing as --num-nodes flag value in gcloud).

Because of that, all the numbers there were incorrect, since they can't be based on those vars.

The description is too long for commit, but I will update PR description.